packages/api/src/external/fhir/shared/hydrate-bundle.ts

```diff
@@ -29,29 +33,62 @@ export function hydrateBundle(
 
 type ReplacementIdPair = { old: string; new: string };
 
-function validateUuidsAndAddExtensions(bundle: ValidBundle, docExtension: Extension): ValidBundle {
+function validateUuidsAndAddExtensions(
+  bundle: ValidBundle,
+  docExtension: Extension,
+  patientId: string
+): ValidBundle {
   const replacements: ReplacementIdPair[] = [];
+  const uniqueIds = new Set<string>();
   bundle.entry.forEach(entry => {
-    const oldId = entry.resource.id;
-    if (!oldId) {
-      entry.resource.id = uuidv7();
-    }
-    if (oldId && !isValidUuid(oldId)) {
-      replacements.push({
-        old: oldId,
-        new: uuidv7(),
-      });
+    const resource: Resource = entry.resource;
+    const oldId = resource.id;
+    if (oldId) {
+      if (uniqueIds.has(oldId)) {
+        throw new BadRequestError(`Multiple resources with the same ID: ${oldId}`);
+      }
+      uniqueIds.add(oldId);
+
+      if (!isValidUuid(oldId)) {
+        replacements.push({
+          old: oldId,
+          new: uuidv7(),
+        });
+      }
     }
+    verifyPatientReferences(resource, patientId);
```

Comment:
So to be clear, this is adding the requirement that the patient id they use in the fhir bundle is the same as the Metriport patient id of the patient. Is that a requirement we explicitly wanted / if it is let's make that explicit in the ticket and the PR descriptions and in customer comms?

Reply:
Ya, this will be included in the docs + CX will quickly learn from the errors we'll throw :P

Reply:
Regarding the ticket and the PR description - it's all already there

Reply:
nit: this doesnt specify the detail I am saying, that not only do all references need to point to the same patient, but they have to point to the patient using the id we assign at patient creation
> We should probably introduce Patient references check to make sure that all resources in a Bundle that are pointing to a Patient are all referring to the same patient. This will help us prevent data cross-contamination between patients due to mistakes on the CX part.